resources/wafv2-apikeys.go

```diff
@@ -0,0 +1,119 @@
+package resources
+
+import (
+	"context"
+
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+
+	"github.com/ekristen/libnuke/pkg/registry"
+	"github.com/ekristen/libnuke/pkg/resource"
+	"github.com/ekristen/libnuke/pkg/types"
+
+	"github.com/ekristen/aws-nuke/v3/pkg/nuke"
+)
+
+const WAFv2APIKeyResource = "WAFv2APIKey" // #nosec G101
+
+func init() {
+	registry.Register(&registry.Registration{
+		Name:   WAFv2APIKeyResource,
+		Scope:  nuke.Account,
+		Lister: &WAFv2APIKeyLister{},
+	})
+}
+
+type WAFv2APIKeyLister struct{}
+
+func (l *WAFv2APIKeyLister) List(_ context.Context, o interface{}) ([]resource.Resource, error) {
+	opts := o.(*nuke.ListerOpts)
+
+	svc := wafv2.New(opts.Session)
+	resources := make([]resource.Resource, 0)
+
+	params := &wafv2.ListAPIKeysInput{
+		Limit: aws.Int64(50),
+		Scope: aws.String("REGIONAL"),
+	}
+
+	output, err := getAPIKeys(svc, params)
+	if err != nil {
+		return []resource.Resource{}, err
+	}
+
+	resources = append(resources, output...)
+
+	if *opts.Session.Config.Region == endpoints.UsEast1RegionID {
+		params.Scope = aws.String("CLOUDFRONT")
+
+		output, err := getAPIKeys(svc, params)
+		if err != nil {
+			return []resource.Resource{}, err
+		}
+
+		resources = append(resources, output...)
+	}
+
+	return resources, nil
+}
+
+func getAPIKeys(svc *wafv2.WAFV2, params *wafv2.ListAPIKeysInput) ([]resource.Resource, error) {
+	resources := make([]resource.Resource, 0)
+	for {
+		resp, err := svc.ListAPIKeys(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, apiKey := range resp.APIKeySummaries {
+			var tokenDomains []string
+			for _, tokenDomain := range apiKey.TokenDomains {
+				tokenDomains = append(tokenDomains, *tokenDomain)
+			}
+			resources = append(resources, &WAFv2APIKey{
+				svc:          svc,
+				apiKey:       apiKey.APIKey,
+				tokenDomains: tokenDomains,
+				scope:        params.Scope,
+			})
+		}
+
+		if resp.NextMarker == nil {
+			break
+		}
+
+		params.NextMarker = resp.NextMarker
+	}
+	return resources, nil
+}
+
+type WAFv2APIKey struct {
+	svc          *wafv2.WAFV2
+	apiKey       *string
+	tokenDomains []string
+	scope        *string
+}
+
+func (f *WAFv2APIKey) Remove(_ context.Context) error {
+	_, err := f.svc.DeleteAPIKey(&wafv2.DeleteAPIKeyInput{
+		APIKey: f.apiKey,
+		Scope:  f.scope,
+	})
+
+	return err
+}
+
+func (f *WAFv2APIKey) String() string {
+	return strings.Join(f.tokenDomains, ", ")
```

Comment:
The `String()` function in this case should probably return the API Key value.

Reply:
I decided against using the API key value in `String()` and `Properties()` because the key is typically 640+ characters and it's unlikely that a user would need to filter based on the API key. Thoughts?

Reply:
Yeah perhaps something like the first 16 characters of the API key could work. That would be enough to help users identify the API key in the console, if necessary. Using token domains for both `String()` and `Properties()` seems redundant. There is also [`CreationTimestamp`](https://docs.aws.amazon.com/sdk-for-go/api/service/wafv2/#APIKeySummary) for each API key which could be useful.

Reply:
It is redundant, but the goal is to eventually remove the `String()` method all together, so I typically put it in both to keep parity. I would adjust String to be the first ~16 characters and add it to properties as well. Might as well at the CreationTimestamp while you are at it.

Reply:
Makes sense. I will work on those changes later today 👍

Reply:
@ekristen see latest commit.